### ISSUE TITLE
[2021] Revisiting Few-sample BERT Fine-tuning

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#133]
## 0. 論文

| タイトル | Revisiting Few-sample BERT Fine-tuning |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2006.05987 https://github.com/asappresearch/revisit-bert-finetuning |
| 著者    | Tianyi Zhang, Felix Wu, Arzoo Katiyar, Kilian Q. Weinberger, Yoav Artzi |
| 投稿日付 | 2020/6/10 on arxiv |

## 1. どんなもの？

BERT などの言語モデルを下位タスクについて fine-tuning するプロセスは今では一般的だが、下位タスクのデータが少ない場合に非常に不安定であることが知られている。
本研究では、少量データでの fine-tuning におけるいくつかの不安定要因を特定し、いくつかの手法により実験的にそれらを削減することに成功した。

## 2. 先行研究と比べてどこがすごい？

BERT の fine-tuning の不安定性について、optimization / initialization / training step から要因を分析し、実際に安定性を改善した。

## 3. 技術や手法のキモはどこ？

- optimization
    - BERTAdam を普通の Adam に変えることで安定性が大きく改善
    - BERTAdam で省かれている、m と v (gradient の 1st, 2nd moment の移動平均) の debias 部分が悪さをしていたっぽい
    - これは学習ステップが大きければ影響は軽減する
- initialization
    - 通常は BERT の output layer 部分のみを新たに initialization するが、transformer-block 部分も top から L ( = 1~6) 層 re-initialization することで安定性、性能ともに改善
    - 初期化した値との L2 loss を見ることで、re-initialization を行ったほうが差分が小さい (良い初期値となっている) ことを確認
    - pre-trained weight は事前学習タスクに過学習していると解釈できる
- training iteration の増加
    - 基本的には 3 epoch での fine-tuning が行われているが、長いほうが安定化 & たまに性能向上
    - どれだけ行うかはデータセット依存っぽい
- 既存の安定化手法での効果を再評価
    - pre-trained weight decay, mixout, layer-wise learning rate decay, intermediate task の4つの手法を再評価
    - BERTAdam -> Adam にすることでどの手法も更に改善したが、オリジナルで主張されているほどの効果はなかっった
    - 多くの手法は改善ではなく、BERTAdam の biased gradient による劣化を抑えていたことが分かる
    - intermediate task や re-init は特に良い効果が確認できた (下表)
    - ![スクリーンショット 2021-06-20 21 33 45](https://user-images.githubusercontent.com/1255485/122674249-3924fd80-d20f-11eb-8363-36c0f3f9901f.png)

## 4. どうやって有効だと検証した？

実験的に安定性を確認した paper なので↑とだいたい同じ。
データセットには GLUE のタスクを使用、特に RTE, MRPC, STS, CoLA のデータをメインで紹介している。

## 5. 議論はある？

- BERT をメインに議論されており、他の pre-train タスクを行ったモデルについては future work
- モデルのアーキテクチャの違いについても future work. RoBERTa 使ったらぜんぜん違う結果に、とかもありうるか？

## 6. 次に読むべき論文は？

- Fine-Tuning Pretrained Language Models: Weight Initializations, Data Orders, and Early Stopping
    - Jesse Dodge, Gabriel Ilharco, Roy Schwartz, Ali Farhadi, Hannaneh Hajishirzi, Noah Smith
    - fine-tuning の不安定性について議論した研究。
    - https://github.com/cfiken/paper-reading/issues/43
